src/sage/combinat/partition.py

```diff
@@ -4201,8 +4201,8 @@ def zero_one_sequence(self):
             sage: all(Partitions().from_zero_one(mu.zero_one_sequence()) == mu for n in range(10) for mu in Partitions(n))
             True
         """
-        tmp = [self[i]-i for i in range(len(self))]
-        return ([Integer(not (i in tmp)) for i in range(-len(self)+1,self.get_part(0)+1)])
+        tmp = [self[i] - i for i in range(len(self))]
```

Comment:
May be `tmp` should be a set and not a list. I'm not sure if it is interesting here, and I agree it's not the object of this PR.